tests/ttnn/unit_tests/operations/test_repeat.py

```diff
@@ -23,3 +23,43 @@ def test_repeat(device):
     output = ttnn.to_torch(output)
 
     assert_with_pcc(torch_result, output, 0.9999)
+
+
+@pytest.mark.parametrize(
+    "input_shape, input_sharded_memory_config_args",
+    [
+        (
+            (1, 2, 128, 128),
+            dict(core_grid=ttnn.CoreGrid(y=4, x=1), strategy=ttnn.ShardStrategy.HEIGHT),
+        ),
+        (
+            (1, 2, 128, 128),
+            dict(core_grid=ttnn.CoreGrid(y=4, x=1), strategy=ttnn.ShardStrategy.WIDTH),
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "repeat_shape",
+    [
+        (1, 2, 1, 1),
+        (1, 4, 1, 1),
+        (1, 1, 4, 1),
+        (4, 1, 1, 1),
+        (1, 1, 1, 4),
+    ],
+)
+def test_repeat_sharded(input_shape, input_sharded_memory_config_args, repeat_shape, device):
+    torch_input_tensor = torch.randn(input_shape, dtype=torch.bfloat16)
+
+    input_sharded_memory_config = ttnn.create_sharded_memory_config(input_shape, **input_sharded_memory_config_args)
+
+    torch_result = torch_input_tensor.repeat(repeat_shape)
+
+    input_tensor = ttnn.from_torch(
+        torch_input_tensor, layout=ttnn.TILE_LAYOUT, device=device, memory_config=input_sharded_memory_config
+    )
+
+    output = ttnn.repeat(input_tensor, ttnn.Shape(repeat_shape))
```

Comment:
can we check how it works when the original shape is padded to align with a tile?

Reply:
unfortunately outside of this op in the creating_sharded infrastructure (before even repeat is called) it only works with shapes with no padding.
This I think will need to be updated when we update the tensor_layout.